core/src/test/scala/unit/kafka/log/UnifiedLogTest.scala

```diff
@@ -952,8 +952,9 @@ class UnifiedLogTest {
     assertEquals(0, lastSeq)
   }
 
-  @Test
-  def testRetentionDeletesProducerStateSnapshots(): Unit = {
+  @ParameterizedTest(name = "testRetentionDeletesProducerStateSnapshots with empty-active-segment: {0}")
+  @ValueSource(booleans = Array(true, false))
+  def testRetentionDeletesProducerStateSnapshots(isEmptyActiveSegment: Boolean): Unit = {
```

Comment:
isEmptyActiveSegment => createEmptyActiveSegment  and change the test name accordingly.